scalding-core/src/main/scala/com/twitter/scalding/reducer_estimation/Common.scala

```diff
@@ -172,17 +172,22 @@ object ReducerEstimatorStepStrategy extends FlowStepStrategy[JobConf] {
     preds: JList[FlowStep[JobConf]],
     step: FlowStep[JobConf]): Unit = {
     val conf = step.getConfig
-    val stepNumReducers = conf.get(Config.HadoopNumReducers)
-
     // whether the reducers have been set explicitly with `withReducers`
     val setExplicitly = conf.getBoolean(Config.WithReducersSetExplicitly, false)
-
-    // log in JobConf what was explicitly set by 'withReducers'
-    if (setExplicitly) conf.set(EstimatorConfig.originalNumReducers, stepNumReducers)
-
     // whether we should override explicitly-specified numReducers
     val overrideExplicit = conf.getBoolean(Config.ReducerEstimatorOverride, false)
 
+    if (setExplicitly && !overrideExplicit) {
+      LOG.info(
+        s"""
+            |Flow step ${step.getName} was configured with reducers
+            |set explicitly (${setExplicitly}) and estimator explicit override
+            |(${overrideExplicit}). Skipping reducer estimation.
+         """.stripMargin)
+      return
```

Comment:
explicit return is generally avoided in scala (makes the code harder to read).

Can we just add an `else { }` for the rest of the code or put that block into a private method and call it there?